### ISSUE TITLE
at-rule-no-unknown

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1515]
Valid:
```css
@media {
  /* Some code */
}
```
Invalid:
```css
@some-strange-at-rule {
  /* Some code */
}
```
With option `ignoreAtRule: ['string', '/regex/']`.

**Comment by jeddy3** [comment]
Sure. 

- Name: `at-rule-no-unknown`
- Primary option: `true`
- Secondary options: `ignoreAtRules: []` (user defined array of custom at-rules)
- Message: "Unexpected unknown at-rule"
- Section: "At-rule"

We're going to have to draw a line in the sand here to define "unknown" though. Else where we've gone with present and past (read: obsolete) spec things being *known*, and preprocessor language extensions and "future" spec stuff as *unknown*. This latter one is ambiguous though and we need to nail it down for this rule as there a [lots](https://tabatkins.github.io/specs/css-when-else/) [of](http://tabatkins.github.io/specs/css-extend-rule/) ["future"](http://tabatkins.github.io/specs/css-apply-rule/) stuff around at-rules.

Anyone got any ideas? Perhaps ["Recommendation"](https://www.w3.org/2005/10/Process-20051014/tr.html#maturity-levels) and above?

As an aside, even when this is label as ready-to-go can we continue to focus on squashing bugs, updating docs, and the [next and `7.0`](https://github.com/stylelint/stylelint/milestones) milestones before adding new features? This also gives the wider community the opportunity to step up and have a go at adding these new features, and give us the time to [support](https://github.com/stylelint/stylelint/pull/1506) [them](https://github.com/stylelint/stylelint/pull/1410) in their endeavour.

**Comment by evilebottnawi** [comment]
@jeddy3 agree

**Comment by evilebottnawi** [comment]
@jeddy3 good question, stylelint should stick `Recomedation` only or not?
/cc @davidtheclark What do you think?

**Comment by jeddy3** [comment]
> If we choise Recommendation, some rules need change behavior (color-hex-length and etc)

I'm not suggesting a stylelint-wide change of behaviour here. I'm suggestion just removing the ambiguity of what we mean by "unknown" in the `*-no-unknown"` rules.

**Comment by evilebottnawi** [comment]
@jeddy3 I think this is a bit excessive fragmentation.

**Comment by jeddy3** [comment]
> Anyone got any ideas? Perhaps "Recommendation" and above?

On second thoughts, the purpose of this rule is to mainly catch spelling mistakes right? And *not* act as a kind of whitelist for certain types of standard-syntax, as we have the dedicated `*-blacklist` and `*-whitelist` rules for that.  Therefore, I think we should make this rule as useful to as many standard-syntax words as possible by being targeted editor's drafts and down. We also make [case-by-case exceptions](https://github.com/stylelint/stylelint/pull/1360) for things like `hgroup` if we think it'll be benefical to the majority of uses. The more standard-syntax words we include in this rule the more useful it is right?

**Comment by evilebottnawi** [comment]
@jeddy3 yep, catch spelling mistakes, and yes we should make this rule as useful to as many standard-syntax words as possible

**Comment by davidtheclark** [comment]
What you've agreed upon seems fine to me.

**Comment by ntwb** [comment]
> What you've agreed upon seems fine to me.

Same, SGTM 👍
